src/Yoanm/Behat3SymfonyExtension/Subscriber/BehatStepLoggerSubscriber.php

```diff
@@ -0,0 +1,176 @@
+<?php
+namespace Yoanm\Behat3SymfonyExtension\Subscriber;
+
+use Behat\Behat\EventDispatcher\Event\BackgroundTested;
+use Behat\Behat\EventDispatcher\Event\ExampleTested;
+use Behat\Behat\EventDispatcher\Event\FeatureTested;
+use Behat\Behat\EventDispatcher\Event\GherkinNodeTested;
+use Behat\Behat\EventDispatcher\Event\OutlineTested;
+use Behat\Behat\EventDispatcher\Event\ScenarioTested;
+use Behat\Behat\EventDispatcher\Event\StepTested;
+use Behat\Gherkin\Node\ExampleNode;
+use Behat\Gherkin\Node\StepContainerInterface;
+use Behat\Gherkin\Node\StepNode;
+use Behat\Testwork\EventDispatcher\Event\AfterTested;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class BehatStepLoggerSubscriber
+ */
+class BehatStepLoggerSubscriber implements EventSubscriberInterface
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    /**
+     * @param LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            FeatureTested::BEFORE => 'featureEvents',
+            FeatureTested::AFTER => 'featureEvents',
+            BackgroundTested::BEFORE => 'backgroundEvents',
+            BackgroundTested::AFTER => 'backgroundEvents',
+            ScenarioTested::BEFORE => 'scenarioEvents',
+            ScenarioTested::AFTER => 'scenarioEvents',
+            OutlineTested::BEFORE => 'outlineEvents',
+            OutlineTested::AFTER => 'outlineEvents',
+            ExampleTested::BEFORE => 'exampleEvents',
+            ExampleTested::AFTER => 'exampleEvents',
+            StepTested::BEFORE => 'stepEvents',
+            StepTested::AFTER => 'stepEvents',
+        ];
+    }
+
+    /**
+     * @param FeatureTested $event
+     */
+    public function featureEvents(FeatureTested $event)
```

Comment:
All function could be merged. It's possible to determine if it's a feature/background/... event with class instance